contracts/HATVault.sol

```diff
@@ -0,0 +1,770 @@
+    }
+
+    /** @notice See {IERC4626Upgradeable-previewRedeem}. */
+    function previewRedeem(uint256 shares) public view virtual override(IERC4626Upgradeable, ERC4626Upgradeable) returns (uint256) {
+        uint256 assets = _convertToAssets(shares, MathUpgradeable.Rounding.Down);
+        uint256 fee = assets * withdrawalFee / HUNDRED_PERCENT;
+        return assets - fee;
+    }
+
+    /* -------------------------------------------------------------------------------- */
+
+    /* --------------------------------- Getters -------------------------------------- */
+
+    /** @notice See {IHATVault-getHATBountySplit}. */
+    function getHATBountySplit() public view returns(IHATVaultsRegistry.HATBountySplit memory) {
+        if (hatBountySplit.governanceHat != NULL_UINT) {
+            return hatBountySplit;
+        } else {
+            (uint256 governanceHat, uint256 hackerHatVested) = registry.defaultHATBountySplit();
+            return IHATVaultsRegistry.HATBountySplit({
+                governanceHat: governanceHat,
+                hackerHatVested: hackerHatVested
+            });
+        }
+    }
+
+    /** @notice See {IHATVault-getArbitrator}. */
+    function getArbitrator() public view returns(address) {
+        if (arbitrator != NULL_ADDRESS) {
+            return arbitrator;
+        } else {
+            return registry.defaultArbitrator();
+        }
+    }
+
+    /** @notice See {IHATVault-getChallengePeriod}. */
+    function getChallengePeriod() public view returns(uint256) {
+        if (challengePeriod != NULL_UINT) {
+            return challengePeriod;
+        } else {
+            return registry.defaultChallengePeriod();
+        }
+    }
+
+    /** @notice See {IHATVault-getChallengeTimeOutPeriod}. */
+    function getChallengeTimeOutPeriod() public view returns(uint256) {
+        if (challengeTimeOutPeriod != NULL_UINT) {
+            return challengeTimeOutPeriod;
+        } else {
+            return registry.defaultChallengeTimeOutPeriod();
+        }
+    }
+
+    /* -------------------------------------------------------------------------------- */
+
+    /* --------------------------------- Helpers -------------------------------------- */
+
+    /**
+    * @dev Deposit funds to the vault. Can only be called if the committee had
+    * checked in and deposits are not paused.
+    * @param caller Caller of the action (msg.sender)
+    * @param receiver Reciever of the shares from the deposit
+    * @param assets Amount of vault's native token to deposit
+    * @param shares Respective amount of shares to be received
+    */
+    function _deposit(
+        address caller,
+        address receiver,
+        uint256 assets,
+        uint256 shares
+    ) internal override virtual nonReentrant {
+        if (!committeeCheckedIn)
+            revert CommitteeNotCheckedInYet();
+        if (shares == 0) revert AmountToDepositIsZero();
```

Comment:
why check shares and not assets? what's wrong if someone wants to deposit a small amount that doesn't amount to shares?

Reply:
Then they basically give assets and get nothing, they won't be able to withdraw the assets deposited...

Reply:
why not? the withdraw is done by assets, not shares.. they can withdraw the same amount they would just not get any rewards, no?